resolwe_bio/tools/bamtobigwig.sh

```diff
@@ -16,29 +17,56 @@ fi
 
 NAME=`basename "${bam_file}" .bam`
 
-time_deeptools=`python -c "import os; print(max(os.path.getsize('${bam_file}') / 1024**3, 1) * ${time_limit})"`
+time=`python -c "import os; print(max(os.path.getsize('${bam_file}') / 1024**3, 1) * ${time_limit})"`
 
-timeout "${time_deeptools}" \
-  bamCoverage \
-    --binSize "${bin_size}" \
-    --bam "${bam_file}" \
-    --outFileName "tmp_${NAME}.bw" \
-    --outFileFormat "bigwig" \
-    --numberOfProcessors "${number_of_cores}"
+if [ "${tool}" == "deeptools" ]
+then
+  timeout "${time}" \
+    bamCoverage \
+      --binSize "${bin_size}" \
+      --bam "${bam_file}" \
+      --outFileName "tmp_${NAME}.bw" \
+      --outFileFormat "bigwig" \
+      --numberOfProcessors "${number_of_cores}"
 
-exit_status="$?"
+  exit_status="$?"
+
+  if [ "${exit_status}" == 124 ]
+  then
+    re-warning "BigWig took too long to process."
+  elif [ "${exit_status}" == 1 ]
+  then
+    re-checkrc "Calculation of BigWig failed."
```

Comment:
`re-checkrc` is not intuitive here. Either use `re-error` or use `re-checkrc` outside of the `if` statement.

It seems to me that the status of the data object should be the same, if BigWig times out or if it fails. If we consider BigWig to be a less important output, we can issue a warning. If we find it important, we can raise an error. But in my view we should do the same thing in both timeout and fail.

Reply:
Thanks Matjaz. I will put `re-warning` in both cases. I have just noticed that the python script ``bigwig_chroms_to_ucsc.py`` will fail in case of warnings because there will be empty input for `infile`. I need to restructure the script.